contracts/lib/Utilities.sol

```diff
@@ -69,18 +75,82 @@ library Utilities {
       length++;
       i = i >> 4;
     }
+
+    if (length > (size*2)) {
+      return "00";
+    }
+
+    bytes memory result = new bytes(size*2);
     // parse byte by byte and construct the string
     i = payload;
     uint mask = 15;
-    bytes memory result = new bytes(length);
-    uint k = length - 1;
+    uint k = (size*2) - 1;
 
     while (i != 0) {
       uint curr = (i & mask);
       result[k--] = curr > 9 ? byte(55 + uint8(curr)) : byte(48 + uint8(curr));
       i = i >> 4;
     }
 
+    uint j;
+    while (j < ((size*2)-length)) {
+      result[j++] = byte(uint8(48));
+    }
+
     return string(result);
   }
+
+  /**
+   * @dev Removes the last element of byte array
+   * @param payload bytes of data
+   * @return byte array without last element
+   */
+  function removeLastElement(
+    bytes memory payload
+  )
+  internal
+  pure
+  returns (
+    bytes memory
+  )
+  {
+    bytes memory output = new bytes(payload.length-1);
+    for (uint i = 0; i<payload.length-1; i++) {
+      output[i] = payload[i];
+    }
+    return output;
+  }
+
+  /**
+ * @dev Extracts public key from signature and payload for consensus signing (signature+transitionValidation)
+ * @param signature bytes of signature
+ * @param docDataRoot bytes32 of signed data
+ * @return byte32 of public key
+ * @return bool true if success, false otherwise
+ */
+  function recoverPublicKeyFromConsensusSignature(
+    bytes memory signature,
+    bytes32 docDataRoot
+  )
+  internal
+  pure
+  returns (
+    bytes32,
+    bool
+  )
+  {
+    if (signature.length != 66) {
+      return (0, false);
```

Comment:
I wanted to execute a `require` here, but in functions that return values do not perform the same way as with no return val. Any suggestions?

Reply:
Can you say more about how to they perform? They return 0 instead of reverting?

Reply:
I noticed that when writing some unit tests, the assertRevert was not triggered. Maybe because was not part of a transaction, as I was testing the code independently. Now that I think about it, the revert should be done as part of the client code that uses this function, not within the function itself, may be too much to assume.